### PR TITLE
Add basic support for Step glyph

### LINKED ISSUE #3644: Would need a Figure.step to create step line in plot
There's actually a step chart but I didn't find a way to simply add a step line to a Figure.

### PR DESCRIPTION
**Pull request by bryevdv**
This PR adds a `Step` glyph (and `Figure.step`) method that supports three modes: `before`, `center`, `after`. No support for hit-testing is included in this work (can be added in follow-on work based on user interest).

Additionally, the hex color formatting for all `examples/reference/models` scripts has been made consistent. 

- [x] issues: fixes #3644
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

@mattpap please confirm the correctness of the TS API changes.

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap the JS API tests specifically are all failing but I have no idea why:

https://travis-ci.org/bokeh/bokeh/jobs/289254272

**Comment by mattpap** [comment]
There are compilation errors: https://travis-ci.org/bokeh/bokeh/jobs/289254272#L1662. Unfortunately, sometimes our build doesn't fail if there are compilation errors.

**Comment by bryevdv** [comment]
OK I think the lat commit should fix that

**Comment by bryevdv** [comment]
@mattpap I will plan to merge this in a few hours barring any comments. It's pretty bare bones but it's fine as a start.

File: bokehjs/src/coffee/api/typings/types.d.ts [hunk 1]
```diff
@@ -8,4 +8,5 @@ declare namespace Bokeh {
   export type FontSize = string;
   export type Map<T> = {[key: string]: T};
   export type JsObj = Map<any>;
+  export type StepType = "after" | "before" | "center";
```
Comment (mattpap) [inline bokehjs/src/coffee/api/typings/types.d.ts]: This should go to `enums.d.ts`.

File: bokehjs/src/coffee/models/glyphs/step.coffee [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    if L < 2
+      return
+
+    ctx.beginPath()
+
+    ctx.moveTo(sx[0], sy[0])
+
+    for i in [1...L]
+
+      if @model.mode == "before"
```
Comment (mattpap) [inline bokehjs/src/coffee/models/glyphs/step.coffee]: `switch` would be preferred over this.

**Comment by bryevdv** [comment]
@mattpap thanks I have made the changes

**Pull request merged**